src/components/SearchBar.js

```diff
@@ -0,0 +1,96 @@
+import { getSearchData } from "utils/data";
+import { dispatch } from "utils/dispatch";
+
+export class SearchBar {
+  searchBar = document.getElementById("search-bar");
+  autoCompleteElement = document.getElementById("search-bar-autocomplete");
+
+  constructor() {
+    this.searchBar.addEventListener("input", () =>
+      this.autoComplete(this.searchBar.value)
+    );
+    this.searchBar.addEventListener("blur", () => {
+      dispatch.call("remove-autocompete-dropdown", this);
+    });
+    this.searchBar.addEventListener("focus", () => {
+      this.searchBar.value = "";
+    });
+    this.searchBar.addEventListener("change", (e) => {
+      if (
+        [...document.getElementsByClassName("autocompleteElement")]
+          .map((a) => a.value)
+          .indexOf(this.searchBar.value) >= 0
+      ) {
+        let val = this.searchBar.value;
+        dispatch.call("choose-autocomplete-element", Window.lmap, val);
+      }
+      e.stopPropagation();
+    });
+    dispatch.on("remove-autocompete-dropdown", this.removeAutocomplete);
```

Comment:
@jimboweb just curious as why `dispatch` is needed for calling some of these class methods? Thus far in the code `dispatch` is used for communication across different components, *not within components themselves*. You *should* be able to call class methods from within other class methods without any problems. If the context of `this` is giving you trouble you can do a few things:

1. Do `self = this;` at the top of the class method and use `self` elsewhere in the method to get the correct `this` context, e.g `self.myClassMethod()`

2. Use arrow functions to define your class methods, similar to the example I gave in the `LeafletMap` class.

Adding too many custom events for `dispatch` might confuse other developers as they could end up thinking that these methods are used to communicate between components when in this case they are not.

Reply:
Got rid of the unnecessary dispatches

Reply:
I would agree with @clhenrick comments here.